### ISSUE TITLE
Karabiner does not pick up on any keys through a virtual desktop connection / screensharing

### ISSUE THREAD

**Issue by momomo** [pqrs-org/Karabiner-Elements#3114]
0. On mac0, install karabiner and launch karabiner event viewer 
1. Launch screen sharing from mac1 to mac0 that has karabiner event viewer on. 
2. Type some keys. Eventviewer does not pick up any of it which makes using screensharing more difficult.

**Comment by MuhammedZakir** [comment]
You will have to use higher-level tools for that. E.g. Hammerspoon and BetterTouchTool.

**Comment by momomo** [comment]
Yes, I am using keyboard maestro, but I use both and in this instance karabiner is not detecting what higher level programs are able to, why?

**Comment by MuhammedZakir** [comment]
Because KE works at the lowest level.

```
physical keyboard -> OS -> low-level remapping tool -> virtual keyboard -> OS -> higher-level remapping tool / apps -> apps
```

Here, "high" and "low" is referring to where it sits in the hierarchy, not meaning "high" is better than "low".

**Comment by momomo** [comment]
I am aware of how karabiner works but I am just saying. 

It is normally the higher that are unable to do what karabiner is doing so karabiner should have no issue doing what they are doing and enable support for virtual machines and screensharing software. 

The keyboard could be the internal one or connected one, not the remote one. Currently I am able to send karabiner commands from the remote computer to karabiner, but this will fail on program detection since karabiner is not picking any of the keys up, while maestro is. 

I understand if it does not do any high level things now but I am saying perhaps it should?
